### PR TITLE
feat(linter-cli): add json format output (#1462) 

### PR DESCRIPTION
**Pull request by msdlisper**
https://eslint.org/docs/latest/use/formatters/

```
oxlint --format json
```

Try to add JSON format output for the linter CLI #1462 . It's not perfect at the moment, We can iterate on this and add new formats.
Currently:
```
[
    {
        "file_path":"fixtures/nan.js",
        "messages":[
            {
                "severity":1,
                "message":"eslint(use-isnan): Requires calls to isNaN() when checking for NaN",
                "labels":[
                    {
                        "label":"",
                        "span":{
                            "offset":7,
                            "len":3
                        }
                    }
                ]
            }
        ]
    }
]
```

Compare eslint: 
```
[{
            "filePath": "/var/lib/jenkins/workspace/eslint Release/eslint/fullOfProblems.js",
            "messages": [
                {
                    "ruleId": "no-unused-vars",
                    "severity": 2,
                    "message": "'addOne' is defined but never used.",
                    "line": 1,
                    "column": 10,
                    "nodeType": "Identifier",
                    "messageId": "unusedVar",
                    "endLine": 1,
                    "endColumn": 16
                }
             ]
}]
```

### CODE REVIEW

File: crates/oxc_diagnostics/src/service.rs [hunk 1]
```diff
@@ -86,6 +98,53 @@ impl DiagnosticService {
...
+                messages.push(message);
+            }
+            messages_with_path
+                .push(MessageWithPath { file_path: path.to_str().unwrap().to_owned(), messages });
+        }
+
+        let json_str = serde_json::to_string(&messages_with_path).unwrap();
+        self.output.set(json_str.clone());
+        buf_writer.write_all(json_str.as_bytes()).unwrap();
+    }
```
Comment (Boshen) [inline crates/oxc_diagnostics/src/service.rs:146]: I think we can DRY this by implementing a `ReportHandler` enum for `GraphicalReportHandler` and `JSONReportHandler`.

We can then configure the service with different reporters, what do you think?

Comment (msdlisper) [reply crates/oxc_diagnostics/src/service.rs:146]: I agree, and I will process

**Comment by Boshen** [comment]
I need to find a time to take a closer look at this. Can I improve upon your PR when I find the time to do so?

**Comment by msdlisper** [comment]
@Boshen Of course :)

**Comment by Boshen** [comment]
Sorry for the late reply. I took a brief look at this and I find it difficult to follow the abstraction around switching report handlers - given that we may add more report handlers in the future per https://eslint.org/docs/latest/use/formatters/#example-source

I may need to architect this a little bit so that we can add and switch report handlers easily.

**Comment by msdlisper** [comment]
Sorry, my mistake caused this PR to shut down😂,

**Comment by Boshen** [comment]
Handled in https://github.com/oxc-project/oxc/pull/1758

**Pull request closed without merging**